src/lib389/cli/dscontainer

```diff
@@ -24,6 +24,7 @@
 #
 
 import grp
+import psutil
```

Comment:
I'd prefer not to add additional dependencies to external packages. `os.kill` can be used to send signals.

Reply:
I'm mainly using psutil because it returns a [Process given a PID](https://github.com/slominskir/389-ds-base/blob/65a113e9cdff8af918032a0d21beb04a78852461/src/lib389/cli/dscontainer#L432).   Is there a good alternative?

Reply:
`ns-slapd` doesn't start new processes, only new threads. So if you send SIGTERM to it, it should stops the threads. 

```
podman top --latest user, pid,  args
USER        PID         COMMAND
root        1           /usr/bin/python3 /usr/libexec/dirsrv/dscontainer -r
root        17          /usr/sbin/ns-slapd -D /etc/dirsrv/slapd-localhost -i /data/run/slapd-localhost.pid
```

I think `os.kill()` should be sufficient, is it not?

Reply:
I guess so, if dscontainer will only ever be 1 process with no children.    I was wondering if I was over-thinking this.   Should we still attempt graceful shutdown SIGTERM, wait, followed by SIGKILL if necessary?  Or is that unnecessary?

Reply:
Existing code in `dscontainer` catches SIGTERM, SIGINT, SIGHUP:
https://github.com/389ds/389-ds-base/blob/1f95b57fb8b03f72c7d2a2ce5b14eed5ba320407/src/lib389/cli/dscontainer#L431-L434

and handles them in `kill_ds()`:
https://github.com/389ds/389-ds-base/blob/1f95b57fb8b03f72c7d2a2ce5b14eed5ba320407/src/lib389/cli/dscontainer#L346-L363 

(`terminate` is a SIGTERM)

So I don't think we need to complicate things here. Container engines do what you describe already:
https://docs.podman.io/en/latest/markdown/podman-stop.1.html
https://docs.docker.com/engine/reference/commandline/container_stop/

Reply:
OK, sounds like we can simplify.  Will update code to send SIGTERM to ns-slapd via os.kill().

Reply:
Actually, as I dug into the code to make the simplification I realized that one major issue with removing psutil and a Process handle is there appears to be no robust way to wait on the process we're trying to kill.   It might be "good enough", but ideally we do signal then wait.  Might be alternative way to obtain process handle to wait on?   Simply doing a sleep and hoping for the best is really no better than the ps, grep, cut, kill hack I have now.

Reply:
Also, it might be better to error on the side of overly cautious, especially since dscontainer may actually be running as PID 1 in some cases and unfortunately that brings an extra burden of carefully cleaning up processes.   Specifically we had discussed [possibly adding a docker-entrypoint-initdb](https://github.com/389ds/389-ds-base/discussions/6058) functionality to dscontainer.  At that point dscontainer would be running user scripts, which could be issuing execs (spawning child processes).

Reply:
Could use https://docs.python.org/3/library/os.html#os.waitpid I suppose.  Is this good enough?  Are we concerned about init (PID 1) use-cases?   Might be separate issue I guess.

Reply:
I think we already have some waitpid code in the dscontainer file you can use?